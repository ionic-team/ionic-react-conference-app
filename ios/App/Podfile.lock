PODS:
  - Capacitor (7.0.1):
    - CapacitorCordova
  - CapacitorCordova (7.0.1)
  - CapacitorPreferences (7.0.0):
    - Capacitor

DEPENDENCIES:
  - "Capacitor (from `../../node_modules/@capacitor/ios`)"
  - "CapacitorCordova (from `../../node_modules/@capacitor/ios`)"
  - "CapacitorPreferences (from `../../node_modules/@capacitor/preferences`)"

EXTERNAL SOURCES:
  Capacitor:
    :path: "../../node_modules/@capacitor/ios"
  CapacitorCordova:
    :path: "../../node_modules/@capacitor/ios"
  CapacitorPreferences:
    :path: "../../node_modules/@capacitor/preferences"

SPEC CHECKSUMS:
  Capacitor: de199cba6c8b20995428ad0b7cb0bc6ca625ffd4
  CapacitorCordova: 63d476958d5022d76f197031e8b7ea3519988c64
  CapacitorPreferences: 8e4bef1f937fc991a19d7ea99330c36a62edd24b

PODFILE CHECKSUM: bc4412675d30bc194ab5961211929e1e7de79579

COCOAPODS: 1.16.2
