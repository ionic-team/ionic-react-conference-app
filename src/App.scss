/*
 * App Global CSS
 * ----------------------------------------------------------------------------
 * Put style rules here that you want to apply globally. These styles are for
 * the entire app and not just one component.
 */
