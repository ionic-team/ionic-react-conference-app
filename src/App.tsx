import React, { useEffect } from 'react';
import { Redirect, Route } from 'react-router-dom';
import { IonApp, IonRouterOutlet, IonSplitPane } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';

import Menu from './components/Menu';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/padding.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/display.css';

/* Theme variables */
import './theme/variables.css';
import MainTabs from './pages/MainTabs';
import { connect } from './data/connect';
import { AppContextProvider } from './data/AppContext';
import { loadConfData } from './data/sessions/sessions.actions';
import { setIsLoggedIn, setUsername, loadUserData } from './data/user/user.actions';
import Account from './pages/Account';
import Login from './pages/Login';
import Signup from './pages/Signup';
import Support from './pages/Support';
import Tutorial from './pages/Tutorial';
import HomeOrTutorial from './components/HomeOrTutorial';
import { Schedule } from "./models/Schedule";

const App: React.FC = () => {
  return (
    <AppContextProvider>
      <IonicAppConnected />
    </AppContextProvider>
  );
};

interface StateProps {
  darkMode: boolean;
  schedule: Schedule;
}

interface DispatchProps {
  loadConfData: typeof loadConfData;
  loadUserData: typeof loadUserData;
  setIsLoggedIn: typeof setIsLoggedIn;
  setUsername: typeof setUsername;
}

interface IonicAppProps extends StateProps, DispatchProps { }

const IonicApp: React.FC<IonicAppProps> = ({ darkMode, schedule, setIsLoggedIn, setUsername, loadConfData, loadUserData }) => {

  useEffect(() => {
    loadUserData();
    loadConfData();
    // eslint-disable-next-line
  }, []);

  return (
    schedule.groups.length === 0 ? (
      <div></div>
    ) : (
        <IonApp className={`${darkMode ? 'dark-theme' : ''}`}>
          <IonReactRouter>
            <IonSplitPane contentId="main">
              <Menu />
              <IonRouterOutlet id="main">
                <Route path="/tabs" component={MainTabs} />
                <Route path="/account" component={Account} />
                <Route path="/login" component={Login} />
                <Route path="/signup" component={Signup} />
                <Route path="/support" component={Support} />
                <Route path="/tutorial" component={Tutorial} />
                <Route path="/logout" render={() => {
                  setIsLoggedIn(false);
                  setUsername(undefined);
                  return <Redirect to="/tabs" />
                }} />
                <Route path="/" component={HomeOrTutorial} exact />
              </IonRouterOutlet>
            </IonSplitPane>
          </IonReactRouter>
        </IonApp>
      )
  )
}

export default App;

const IonicAppConnected = connect<{}, StateProps, DispatchProps>({
  mapStateToProps: (state) => ({
    darkMode: state.user.darkMode,
    schedule: state.data.schedule
  }),
  mapDispatchToProps: { loadConfData, loadUserData, setIsLoggedIn, setUsername },
  component: IonicApp
});
