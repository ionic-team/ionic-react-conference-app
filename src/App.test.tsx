import App from './App';
import { render } from '@testing-library/react';

it('renders without crashing', () => {
  const { asFragment, container } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
