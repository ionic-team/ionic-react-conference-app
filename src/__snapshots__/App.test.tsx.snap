// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <ion-app>
    <ion-split-pane
      content-id="main"
    >
      <ion-menu
        content-id="main"
      >
        <ion-header>
          <ion-toolbar>
            <ion-title>
              Menu
            </ion-title>
          </ion-toolbar>
        </ion-header>
        <ion-content
          class="outer-content"
        >
          <ion-list>
            <ion-list-header>
              Navigate
            </ion-list-header>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  Schedule
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  Speakers
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  Map
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  About
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
          </ion-list>
          <ion-list>
            <ion-list-header>
              Account
            </ion-list-header>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  Account
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  Support
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
            <ion-menu-toggle
              auto-hide="false"
            >
              <ion-item>
                <ion-icon
                  slot="start"
                />
                <ion-label>
                  Logout
                </ion-label>
              </ion-item>
            </ion-menu-toggle>
          </ion-list>
          <ion-list>
            <ion-list-header>
              Tutorial
            </ion-list-header>
            <ion-item>
              <ion-icon
                slot="start"
              />
              Show Tutorial
            </ion-item>
          </ion-list>
        </ion-content>
      </ion-menu>
      <ion-router-outlet
        id="main"
      >
        <div
          style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
        >
          <div
            class="tabs-inner"
            style="position: relative; flex: 1; contain: layout size style;"
          >
            <ion-router-outlet>
              <div
                class="ion-page ion-page-invisible"
              >
                <ion-header>
                  <ion-toolbar
                    color="primary"
                  >
                    <ion-buttons
                      slot="start"
                    >
                      <ion-menu-button />
                    </ion-buttons>
                    <ion-segment>
                      <ion-segment-button
                        value="all"
                      >
                        All
                      </ion-segment-button>
                      <ion-segment-button
                        value="toSync"
                      >
                        Favorites
                      </ion-segment-button>
                    </ion-segment>
                    <ion-buttons
                      slot="end"
                    >
                      <ion-button>
                        <ion-icon
                          slot="icon-only"
                        />
                      </ion-button>
                    </ion-buttons>
                  </ion-toolbar>
                  <ion-toolbar
                    color="primary"
                  >
                    <ion-searchbar
                      placeholder="Search"
                    />
                  </ion-toolbar>
                </ion-header>
                <ion-content>
                  <ion-refresher
                    slot="fixed"
                  >
                    <ion-refresher-content />
                  </ion-refresher>
                  <ion-list>
                    <ion-list-header>
                      No Sessions Found
                    </ion-list-header>
                  </ion-list>
                  <ion-list
                    style="display: none;"
                  />
                </ion-content>
                <ion-fab
                  horizontal="end"
                  slot="fixed"
                  vertical="bottom"
                >
                  <ion-fab-button>
                    <ion-icon />
                  </ion-fab-button>
                  <ion-fab-list
                    side="top"
                  >
                    <ion-fab-button
                      color="vimeo"
                    >
                      <ion-icon />
                    </ion-fab-button>
                    <ion-fab-button
                      color="google"
                    >
                      <ion-icon />
                    </ion-fab-button>
                    <ion-fab-button
                      color="twitter"
                    >
                      <ion-icon />
                    </ion-fab-button>
                    <ion-fab-button
                      color="facebook"
                    >
                      <ion-icon />
                    </ion-fab-button>
                  </ion-fab-list>
                </ion-fab>
              </div>
            </ion-router-outlet>
          </div>
          <ion-tab-bar
            current-path="/tabs/schedule"
            selected-tab="schedule"
            slot="bottom"
          >
            <ion-tab-button
              href="/tabs/schedule"
              tab="schedule"
            >
              <ion-icon />
              <ion-label>
                Schedule
              </ion-label>
            </ion-tab-button>
            <ion-tab-button
              href="/tabs/speakers"
              tab="speakers"
            >
              <ion-icon />
              <ion-label>
                Speakers
              </ion-label>
            </ion-tab-button>
            <ion-tab-button
              href="/tabs/map"
              tab="map"
            >
              <ion-icon />
              <ion-label>
                Map
              </ion-label>
            </ion-tab-button>
            <ion-tab-button
              href="/tabs/about"
              tab="about"
            >
              <ion-icon />
              <ion-label>
                About
              </ion-label>
            </ion-tab-button>
          </ion-tab-bar>
        </div>
      </ion-router-outlet>
    </ion-split-pane>
  </ion-app>
</DocumentFragment>
`;
